### PR TITLE
ignition: add boot-process document

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

File: ignition/boot-process.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# CoreOS startup process
+
+The CoreOS startup process is fundamentally built on the standard [Linux startup process][linux startup]. Since this process is already well documented and generally well understood, this document will focus on particular aspects specific to CoreOS's startup process.
```
Comment (joshix) [inline ignition/boot-process.md]: s/fundamentally//
s/particular//
"specific to booting CoreOS."

File: ignition/boot-process.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Bootloader
+
+The first component of CoreOS to execute on a booting system is [GRUB][grub]. The [GRUB config][grub config] used by CoreOS serves several important roles.
```
Comment (joshix) [inline ignition/boot-process.md]: `[GRUB][grub] is the first program executed when a CoreOS system boots.`

File: ignition/boot-process.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+First off, it [determines which USR partition to use][gptprio.next] which is what allows CoreOS to do its automatic, atomic upgrades and rollbacks.
```
Comment (joshix) [inline ignition/boot-process.md]:
```
The CoreOS [GRUB config][grubconfig] has several roles. First, the GRUB config [specifies which `usr` partition to use][gptprio.next] from the two `usr` partitions CoreOS uses to provide atomic upgrades and rollbacks.
```

File: ignition/boot-process.md (hunk 2 above)
Comment (joshix) [inline ignition/boot-process.md]: Strike second sentence; replaced in line 9 below

**Comment by crawford** [comment]
Updated. This doc is a little half-assed right now. I think it's fine in regard to Ignition, but I think detailing a lot more of the OS would be helpful. And if we went that route, I don't think this would live in the Ignition section.

File: ignition/boot-process.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+The CoreOS [GRUB config][grubconfig] has several roles. First, the GRUB config [specifies which `usr` partition to use][gptprio.next] from the two `usr` partitions CoreOS uses to provide atomic upgrades and rollbacks.
+
+Additionally, GRUB determines if this is the first time a machine has booted. This is accomplished by [searching for the initial disk GUID][search] (00000000-0000-0000-0000-000000000001). CoreOS is built with this well-known disk GUID in order to detect the first boot. This GUID is randomized later in the boot process so that individual disks may be uniquely identified. If GRUB detects that this is in fact a first boot, it sets two Linux kernel command line parameters: `coreos.randomize_guid=00000000-0000-0000-0000-000000000001` and `coreos.first_boot=1`. These parameters are used by various programs in later stages of the boot process.
+
+## Early user space
+
+The next major milestone in the CoreOS startup process is the jump into the initial RAM file system. In addition to its standard responsibility of mounting the root filesystem, the initramfs is also where the disk GUID is randomized and where Ignition runs.
```
Comment (robszumski) [inline ignition/boot-process.md]: Probably need to work in "sometimes referred to as the initrd" somewhere

Comment (crawford) [inline ignition/boot-process.md]: The initrd and initramfs are actually different things. They serve the same purpose, but their semantics and format are different (initrd is legacy). If people recognize "initrd", I'm pretty sure they will recognize "initramfs" as well. Maybe not though.

**Comment by joshix** [comment]
LGTM

**Pull request merged**